tests/unit/testsuite/Bolt/Boltpay/Model/BoltOrderTest.php

```diff
@@ -331,4 +331,35 @@ public function testCorrectBillingAddressWithNoBillingAddress() {
 
         $this->assertEquals($expected, $result);
     }
-}
\ No newline at end of file
+
+    /**
+     * @test
+     * @group Discount
+     * @dataProvider addDiscountsCases
+     */
+    public function addDiscounts($data = [])
+    {
+        
+        $cart = $this->testHelper->addProduct(self::$productId, 2);
+        $quote = $cart->getQuote();
+        $totals = $quote->getTotals();
+        
+        $class = new \ReflectionClass($this->currentMock);
+        $method = $class->getMethod('addDiscounts');
+        $method->setAccessible(true);
+        $totalDiscount = $method->invokeArgs($this->currentMock, [$totals, []]);
+        $this->assertEquals($data['expect'], $totalDiscount);
+        $this->assertTrue(true);
+    }
+
+    public function addDiscountsCases()
```

Comment:
Please add method comment describing this methods use. It appears to be an internal method, so would better be set to private. If it is a utility method for wider use than this test, then it is best placed in a helper class.